numpy/random/mtrand/generate_randint_helpers.py

```diff
@@ -0,0 +1,152 @@
+"""
+Helper file for generating the necessary helper functions
+for each `dtype` in `np.random.randint`. The function
+`gen_randint_helpers` is called right before Cythonization
+in NumPy setup.
+
+"""
+
+# npy_dt : Desired C API data type
+# npy_udt : Unsigned version of desired C
+#           API data type (could be identical)
+# npy_dt : Desired Python API data type
+
+template = """
+def _rand_{npy_dt}(low, high, size, rngstate):
```

Comment:
Yeah...I was just thinking about that.  I'm hesitant to do such a refactoring, as I don't feel it is within the scope of this PR.  Also, the only re-implementation I can see at this point is doing `if`-`else` as you did <a href="https://github.com/bashtage/ng-numpy-randomstate/blob/master/randomstate/randomstate.pyx#L849">here</a>, but I think that's something we're trying to avoid in the `numpy` codebase.